src/Microsoft.OData.Core/UriParser/Binders/LiteralBinder.cs

```diff
@@ -22,33 +22,46 @@ internal static QueryNode BindLiteral(LiteralToken literalToken)
         {
             ExceptionUtils.CheckArgumentNotNull(literalToken, "literalToken");
 
+            if (!string.IsNullOrEmpty(literalToken.OriginalText))
+            {
+                if (literalToken.ExpectedEdmTypeReference != null)
+                {
+                    return new ConstantNode(literalToken.Value, literalToken.OriginalText, literalToken.ExpectedEdmTypeReference);
+                }

```

Comment:
Do we need to do this check, or can we just pass literalToken.ExpectedEdmTypeReference to the ConstantNode constructor, even if it's null?

Reply:
Yes. We need this check. Explicitly setting the expected type to null causes problems. I can share the details if you are interested but in short, this check is necessary.